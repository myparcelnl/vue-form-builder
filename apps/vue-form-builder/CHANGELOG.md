# Changelog

<!-- MONODEPLOY:BELOW -->

## [1.0.0-beta.39](https://github/myparcelnl/vue-form-builder/compare/@myparcel/vue-form-builder@1.0.0-beta.38...@myparcel/vue-form-builder@1.0.0-beta.39) "@myparcel/vue-form-builder" (2024-01-25)


### Bug Fixes

* **bundle:** improve tree-shaking ([be1ae30](https://github/myparcelnl/vue-form-builder/commit/be1ae3089cdad141a6b7f65fa0dfc4dcc0720876))




## [1.0.0-beta.36](https://github/myparcelnl/vue-form-builder/compare/@myparcel/vue-form-builder@1.0.0-beta.35...@myparcel/vue-form-builder@1.0.0-beta.36) "@myparcel/vue-form-builder" (2024-01-10)




## [1.0.0-beta.35](https://github/myparcelnl/vue-form-builder/compare/@myparcel/vue-form-builder@1.0.0-beta.34...@myparcel/vue-form-builder@1.0.0-beta.35) "@myparcel/vue-form-builder" (2024-01-05)


### Features

* **types:** export ComponentProps type ([02e27f4](https://github/myparcelnl/vue-form-builder/commit/02e27f4f9cf6921d2d5489b638bbaf9b7dd7a8d3))




## [1.0.0-beta.29](https://github/myparcelnl/vue-form-builder/compare/@myparcel/vue-form-builder@1.0.0-beta.28...@myparcel/vue-form-builder@1.0.0-beta.29) "@myparcel/vue-form-builder" (2023-12-07)




## [1.0.0-beta.28](https://github/myparcelnl/vue-form-builder/compare/@myparcel/vue-form-builder@1.0.0-beta.27...@myparcel/vue-form-builder@1.0.0-beta.28) "@myparcel/vue-form-builder" (2023-12-07)


### ⚠ BREAKING CHANGES

* `vue-form-builder-js` is removed. `vue-form-builder` is now compiled JavaScript instead of TypeScript. Import `@myparcel/vue-form-builder/ts` to get TypeScript

### Build System

* merge vue-form-builder-js into vue-form-builder ([e10c874](https://github/myparcelnl/vue-form-builder/commit/e10c874a3fc6bb5ddeef4ec68c28c322cc04d191))




## [1.0.0-beta.27](https://github/myparcelnl/vue-form-builder/compare/@myparcel/vue-form-builder@1.0.0-beta.26...@myparcel/vue-form-builder@1.0.0-beta.27) "@myparcel/vue-form-builder" (2023-12-06)


### Bug Fixes

* fix circular imports ([0d4a4d4](https://github/myparcelnl/vue-form-builder/commit/0d4a4d43d4bd62b922dedeab4b965bd27105c693))




## [1.0.0-beta.18](https://github/myparcelnl/vue-form-builder/compare/@myparcel/vue-form-builder@1.0.0-beta.17...@myparcel/vue-form-builder@1.0.0-beta.18) "@myparcel/vue-form-builder" (2023-11-21)


### Features

* add new way of declaring forms ([#218](https://github/myparcelnl/vue-form-builder/issues/218)) ([9546c82](https://github/myparcelnl/vue-form-builder/commit/9546c82d29716cc269fe6731ff682f0670beec7a))




## [1.0.0-beta.8](///compare/@myparcel/vue-form-builder@1.0.0-beta.7...@myparcel/vue-form-builder@1.0.0-beta.8) "@myparcel/vue-form-builder" (2023-06-05)




## [1.0.0-beta.7](///compare/@myparcel/vue-form-builder@1.0.0-beta.6...@myparcel/vue-form-builder@1.0.0-beta.7) "@myparcel/vue-form-builder" (2023-05-31)


### ⚠ BREAKING CHANGES

* **meta:** fix having to use /src in vfb imports (#208)

### Bug Fixes

* **meta:** fix having to use /src in vfb imports ([#208](///issues/208)) 2e52261




## [1.0.0-beta.5](///compare/@myparcel/vue-form-builder@1.0.0-beta.4...@myparcel/vue-form-builder@1.0.0-beta.5) "@myparcel/vue-form-builder" (2023-05-30)


### Bug Fixes

* include .d.ts files in final package 12a9b7b




## [1.0.0-beta.4](https://github/myparcelnl/vue-form-builder/compare/@myparcel/vue-form-builder@1.0.0-beta.3...@myparcel/vue-form-builder@1.0.0-beta.4) "@myparcel/vue-form-builder" (2023-05-30)


### Bug Fixes

* add FormHook enum to exports ([364732d](https://github/myparcelnl/vue-form-builder/commit/364732d580518d77304b393d03ab37f47cf64c68))




## [1.0.0-beta.3](https://github/myparcelnl/vue-form-builder/compare/@myparcel/vue-form-builder@1.0.0-beta.2...@myparcel/vue-form-builder@1.0.0-beta.3) "@myparcel/vue-form-builder" (2023-05-23)




## [1.0.0-beta.1](https://github/myparcelnl/vue-form-builder/compare/@myparcel/vue-form-builder@1.0.0-beta.0...@myparcel/vue-form-builder@1.0.0-beta.1) "@myparcel/vue-form-builder" (2023-05-04)




## [1.0.0-beta.0](https://github/myparcelnl/vue-form-builder/compare/@myparcel/vue-form-builder@1.0.0-alpha.1...@myparcel/vue-form-builder@1.0.0-beta.0) "@myparcel/vue-form-builder" (2023-04-06)


### Features

* add src to module files ([c14626a](https://github/myparcelnl/vue-form-builder/commit/c14626a2ab1c98464611f83978575a2ce84c53a2))
* expose useForm and useElement ([a9df264](https://github/myparcelnl/vue-form-builder/commit/a9df2643ead2bd9ca976f1a158761acb4c34cf83))
* **plugin:** add createMyParcelFormBuilderPlugin ([af207cd](https://github/myparcelnl/vue-form-builder/commit/af207cd1cc1810484b08386ce259f3fd9dce5d2e))
* separate apps and libs ([1bc04c7](https://github/myparcelnl/vue-form-builder/commit/1bc04c7625e0036bb3d72c40f471902e8232ce71))


### Bug Fixes

* export vue injections ([b95dfc6](https://github/myparcelnl/vue-form-builder/commit/b95dfc6a7a12eb654d7ef8d976689ee56862cc9e))
* remove reactive ([9edf168](https://github/myparcelnl/vue-form-builder/commit/9edf168e5499a6d129e5dcaac818c4e3fc1bce99))




## [1.0.0-alpha.44](https://github/myparcelnl/vue-form-builder/compare/@myparcel/vue-form-builder@1.0.0-alpha.43...@myparcel/vue-form-builder@1.0.0-alpha.44) "@myparcel/vue-form-builder" (2023-03-28)




## [1.0.0-alpha.44](https://github/myparcelnl/vue-form-builder/compare/@myparcel/vue-form-builder@1.0.0-alpha.43...@myparcel/vue-form-builder@1.0.0-alpha.44) "@myparcel/vue-form-builder" (2023-03-17)

## [1.0.0-alpha.38](https://github/myparcelnl/vue-form-builder/compare/@myparcel/vue-form-builder@1.0.0-alpha.37...@myparcel/vue-form-builder@1.0.0-alpha.38) "@myparcel/vue-form-builder" (2023-02-22)

### Features

- expose useForm and useElement ([a9df264](https://github/myparcelnl/vue-form-builder/commit/a9df2643ead2bd9ca976f1a158761acb4c34cf83))

## [1.0.0-alpha.35](https://github/myparcelnl/vue-form-builder/compare/@myparcel/vue-form-builder@1.0.0-alpha.34...@myparcel/vue-form-builder@1.0.0-alpha.35) "@myparcel/vue-form-builder" (2023-02-17)

## [1.0.0-alpha.34](https://github/myparcelnl/vue-form-builder/compare/@myparcel/vue-form-builder@1.0.0-alpha.33...@myparcel/vue-form-builder@1.0.0-alpha.34) "@myparcel/vue-form-builder" (2023-02-16)

## [1.0.0-alpha.33](https://github/myparcelnl/vue-form-builder/compare/@myparcel/vue-form-builder@1.0.0-alpha.32...@myparcel/vue-form-builder@1.0.0-alpha.33) "@myparcel/vue-form-builder" (2023-02-07)

### Features

- add src to module files ([c14626a](https://github/myparcelnl/vue-form-builder/commit/c14626a2ab1c98464611f83978575a2ce84c53a2))

## [1.0.0-alpha.32](https://github/myparcelnl/vue-form-builder/compare/@myparcel/vue-form-builder@1.0.0-alpha.31...@myparcel/vue-form-builder@1.0.0-alpha.32) "@myparcel/vue-form-builder" (2023-01-31)

## [1.0.0-alpha.30](https://github/myparcelnl/vue-form-builder/compare/@myparcel/vue-form-builder@1.0.0-alpha.29...@myparcel/vue-form-builder@1.0.0-alpha.30) "@myparcel/vue-form-builder" (2023-01-19)

## [1.0.0-alpha.29](https://github/myparcelnl/vue-form-builder/compare/@myparcel/vue-form-builder@1.0.0-alpha.28...@myparcel/vue-form-builder@1.0.0-alpha.29) "@myparcel/vue-form-builder" (2023-01-09)

### Features

- **plugin:** add createMyParcelFormBuilderPlugin ([af207cd](https://github/myparcelnl/vue-form-builder/commit/af207cd1cc1810484b08386ce259f3fd9dce5d2e))

## [1.0.0-alpha.27](https://github/myparcelnl/vue-form-builder/compare/@myparcel/vue-form-builder@1.0.0-alpha.26...@myparcel/vue-form-builder@1.0.0-alpha.27) "@myparcel/vue-form-builder" (2023-01-05)

### Bug Fixes

- remove reactive ([9edf168](https://github/myparcelnl/vue-form-builder/commit/9edf168e5499a6d129e5dcaac818c4e3fc1bce99))

## [1.0.0-alpha.26](https://github/myparcelnl/vue-form-builder/compare/@myparcel/vue-form-builder@1.0.0-alpha.25...@myparcel/vue-form-builder@1.0.0-alpha.26) "@myparcel/vue-form-builder" (2023-01-04)

## [1.0.0-alpha.24](https://github/myparcelnl/vue-form-builder/compare/@myparcel/vue-form-builder@1.0.0-alpha.23...@myparcel/vue-form-builder@1.0.0-alpha.24) "@myparcel/vue-form-builder" (2022-12-20)

### Bug Fixes

- export vue injections ([b95dfc6](https://github/myparcelnl/vue-form-builder/commit/b95dfc6a7a12eb654d7ef8d976689ee56862cc9e))

## [1.0.0-alpha.23](https://github/myparcelnl/vue-form-builder/compare/@myparcel/vue-form-builder@1.0.0-alpha.22...@myparcel/vue-form-builder@1.0.0-alpha.23) "@myparcel/vue-form-builder" (2022-12-19)

### Features

- separate apps and libs ([1bc04c7](https://github/myparcelnl/vue-form-builder/commit/1bc04c7625e0036bb3d72c40f471902e8232ce71))
