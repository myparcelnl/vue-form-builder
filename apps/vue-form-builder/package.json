{
  "name": "@myparcel/vue-form-builder",
  "version": "1.0.0-beta.1",
  "keywords": [
    "vue",
    "form-builder",
    "form",
    "vue3"
  ],
  "repository": "github:myparcelnl/vue-form-builder",
  "license": "MIT",
  "type": "module",
  "exports": {
    ".": "./dist/index.js",
    "./src": "./src/index.ts"
  },
  "main": "./dist/index.js",
  "types": "./dist/index.d.ts",
  "files": [
    "./dist",
    "./src"
  ],
  "scripts": {
    "build": "run ws:tsup:build \"$(pwd)\"",
    "build:dev": "run ws:tsup:build:dev \"$(pwd)\"",
    "clean": "run ws:clean \"$(pwd)\"",
    "serve": "run ws:tsup:serve \"$(pwd)\"",
    "typecheck": "run ws:tsup:typecheck \"$(pwd)\"",
    "watch": "run ws:tsup:watch \"$(pwd)\""
  },
  "dependencies": {
    "@myparcel-vfb/core": "workspace:^1.0.0-beta.1",
    "@myparcel-vfb/hook-manager": "workspace:^1.0.0-beta.1",
    "@myparcel-vfb/plugin": "workspace:^1.0.0-beta.1",
    "@myparcel-vfb/utils": "workspace:^1.0.0-beta.1",
    "@myparcel/ts-utils": "^1.6.0"
  },
  "devDependencies": {
    "@myparcel-vfb/build-tsup": "workspace:*",
    "tsup": "^6.6.3",
    "typescript": "^4.9.5",
    "vue": "^3.2.47",
    "vue-tsc": "^1.1.5"
  },
  "publishConfig": {
    "access": "public"
  }
}
