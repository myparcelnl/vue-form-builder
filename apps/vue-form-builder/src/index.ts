export type {
  AnyElementConfiguration,
  AnyElementInstance,
  ComponentOrHtmlElement,
  ElementName,
  FormConfiguration,
  FormInstance,
  InteractiveElementConfiguration,
  InteractiveElementInstance,
  PlainElementConfiguration,
  PlainElementInstance,
  ResolvedElementConfiguration,
  Validator,
} from '@myparcel-vfb/core/src';

export type {SelectOption} from '@myparcel-vfb/utils/src';

export {MagicForm, defineField, defineForm, useElement, useForm, useFormBuilder} from '@myparcel-vfb/core/src';

export {MyParcelFormBuilderPlugin, createMyParcelFormBuilderPlugin} from '@myparcel-vfb/plugin/src';
