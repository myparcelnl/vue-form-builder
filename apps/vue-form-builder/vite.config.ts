import dts from 'vite-plugin-dts';
import {createViteConfig} from '@myparcel-vfb/build-vite';

export default createViteConfig(() => {
  return {
    plugins: [dts({entryRoot: 'src'})],

    build: {
      rollupOptions: {
        external: ['vue', '@myparcel/ts-utils', '@vueuse/core'],
        output: {
          globals: {
            vue: 'Vue',
          },
        },
      },
    },
  };
});
