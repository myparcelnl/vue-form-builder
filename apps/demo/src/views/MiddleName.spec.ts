import {defineComponent, h, nextTick, ref} from 'vue';
import {describe, it, expect, afterEach} from 'vitest';
import {render, waitFor} from '@testing-library/vue';
import userEvent from '@testing-library/user-event';
import {type FormInstance} from '@myparcel/vue-form-builder';
import {createForm} from '@myparcel/vue-form-builder';
import MiddleName from './MiddleName.vue';

/* c8 ignore next */
const scheduler = typeof setImmediate === 'function' ? setImmediate : setTimeout;

/**
 * flushPromises implementation as described in https://github.com/kentor/flush-promises
 * Identical function as is used in vue-test-utils.
 * @returns {Promise<void>}
 */
export const flushPromises = (): Promise<void> => {
  return new Promise((resolve) => {
    scheduler(resolve);
  });
};

describe('Middle Name', () => {
  let Form;

  const mountForm = async () => {
    Form = createForm('middle-name-form', {});

    const FormComponent = defineComponent({
      name: 'MockPage',
      render() {
        return h(Form.Component, () => h(MiddleName));
      },
    });

    const component = render(FormComponent);

    await flushPromises();
    return component;
  };

  it('should render a pre-populated form, transform inserted form fields', async () => {
    const {html} = await mountForm();
    expect(html()).toMatchSnapshot();

    const middleName = Form.instance.getField('middleName');

    console.log('middleName', middleName.isOptional);
  });
});
