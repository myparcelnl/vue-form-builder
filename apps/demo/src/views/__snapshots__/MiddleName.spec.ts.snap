// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Middle Name > should render a pre-populated form, transform inserted form fields 1`] = `
"<form id="middle-name-form">
  <div><input id="middleName" type="text" name="middleName" class="" data-test-id="middleName"></div>
</form>"
`;
